pkg/middleware/requestid/requestid.go

```diff
@@ -13,13 +14,24 @@ import (
 
 const (
 	requestIDCtxKey   = "request-id-context-key"
+	requestIDKey 			= "request_id"
 	requestIDTraceKey = "request_id"
 
 	// RequestIDHeader defines the HTTP header that is set in each HTTP response
 	// for a given request. The value of the header is unique per request.
 	RequestIDHeader = "X-Request-Id"
 )
 
+// InitID returns the ID to be used to identify the request.
+// If trace is enabled, returns trace ID; otherwise returns a new ULID.
+func InitID(ctx context.Context) string {
+	spanCtx := trace.SpanContextFromContext(ctx)
+	if spanCtx.TraceID().IsValid() {
+		return spanCtx.TraceID().String()
+	} 
+	return ulid.Make().String()
```

Comment:
@openfga/backend what do you think, should we use `ulid` or `UUID` for request IDs? From https://github.com/oklog/ulid

> If you don't care about time-based ordering of generated IDs, then there's no reason to use ULIDs! There are many other kinds of IDs that are easier, faster, smaller, etc. Consider UUIDs.

Changing this format to `ulid`:
- is a breaking change for consumers. (However, we already use `ulid` for store IDs)
- we can get rid of the `"github.com/google/uuid"` dependency.
- ULID contains information about the time, which for logging is a benefit.
- [Very low chance of collision](https://pkg.go.dev/github.com/oklog/ulid/v2#section-readme:~:text=1.21e%2B24%20unique%20ULIDs%20per%20millisecond%20(1%2C208%2C925%2C819%2C614%2C629%2C174%2C706%2C176%20to%20be%20exact))

Reply:
While waiting for comments, I keep thinking about the "breaking change for consumers" statement.
So far I can only think of pros and can't think of negative scenarios for consumers to justify keeping UUIDs.

I look forward to learning what OpenFGA team thinks about this.

Thank you

Reply:
I asked the team, everyone agreed that having request IDs that are OTEL UUIDs when tracing is enabled and ULIDs when tracing is disabled is a no-go. For request IDs we need some way of getting a OTEL UIID even when tracing is off. However, the ID generator they use isn't exported (https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/trace/id_generator.go#L60). Their code hasn't changed in years though, so my thought was that we could just ... copy it here. 🤔 🤷‍♀️

Reply:
If ULID vs UUID is still a hotly debated topic, as an alternative to unblock this PR, may I keep UUID to avoid rushing the decision on UUID vs ULID.

@miparnisari / team, any thoughts?

Thank you

Reply:
Yes, let's do that :) 

So to summarize:

```
before this PR, with tracing:  "trace_id": "203d831d32d064b47d746223b92da657", "request_id": "cf5012eb-8b32-4e68-8a85-f296cc1a8a80",

before this PR, without tracing: "request_id": "38fee7ac-4bfe-4cf6-baa2-8b5ec296b485"

after this PR, with tracing: "trace_id": "1e20da43269fe07e3d2ac018c0aad2d1",  "request_id": "1e20da43269fe07e3d2ac018c0aad2d1"
(same values)

after this PR, without tracing: "request_id": "38fee7ac-4bfe-4cf6-baa2-8b5ec296b485"
(uuid)
```